src/crimson/mgr/client.cc

```diff
@@ -105,10 +105,11 @@ seastar::future<> Client::reconnect()
   return seastar::sleep(a_while).then([this] {
     auto peer = [&] {
       auto& mgr_addrs = mgrmap.get_active_addrs();
-      if (msgr.get_myaddr().is_legacy()) {
-        return mgr_addrs.legacy_addr();
-      } else {
+      if (msgr.get_myaddr().is_any() ||
+          msgr.get_myaddr().is_msgr2()) {
```

Comment:
I'm not quite understand the impact from this change.

Seems to me before and after this change, the condition `msgr.get_myaddr().is_any() == true` always lead to `return mgr_addrs.msgr2_addr();`, which looks identical.

Am I missing something?

Reply:
no. they are identical. will close this PR.